multisig/src/lib.rs

```diff
@@ -262,6 +326,85 @@ decl_module! {
 		/// Deposit one of this module's events by using the default implementation.
 		fn deposit_event() = default;
 
+		/// Create a new multisig account.
+		/// TODO: update weights for `register_multisig`
+		/// # <weight>
+		/// Key: M - length of members,
+		/// - One storage reads - O(1)
+		/// - One search in sorted list - O(logM)
+		/// - Confirmation that the list is sorted - O(M)
+		/// - One storage writes - O(1)
+		/// - One event
+		/// Total Complexity: O(M + logM)
+		/// # <weight>
+		#[weight = 1]
+		fn register_multisig(origin, signatories: Vec<T::AccountId>, threshold: Percent) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let block_num = <system::Module<T>>::block_number();
+			let nonce = <system::Module<T>>::account_nonce(&who);
+			let multisig_account_id = Self::multi_account_id(
+				&who,
+				block_num,
+				nonce
+			);
+			ensure!(Self::accounts(&multisig_account_id).is_none(), Error::<T>::MultisigAlreadyExists);
+			ensure!(!threshold.is_zero(), Error::<T>::ZeroThreshold);
+			ensure!(signatories.len() > 1, Error::<T>::TooFewSignatories);
+			let max_sigs = T::MaxSignatories::get() as usize;
+			ensure!(signatories.len() <= max_sigs, Error::<T>::TooManySignatories);
+			ensure!(signatories.contains(&who), Error::<T>::SenderNotInSignatories);
+			ensure!(Self::is_sort_and_unique(&signatories), Error::<T>::SignatoriesAreNotUniqueOrUnordered);
+			let multisig_config = MultisigAccount::new(signatories, threshold);
+			<Accounts<T>>::insert(&multisig_account_id, multisig_config);
+			Self::deposit_event(RawEvent::MultisigAccountCreated(multisig_account_id));
+			Ok(())
+		}
+
+		/// Remove the signatory from the multisig account.
+		/// Can only be called by a multisig account.
+		///
+		/// TODO: update weights for `remove_signatory`
+		/// # <weight>
+		/// Key: length of members in multisigConfig: M
+		/// - One storage reads - O(1)
+		/// - remove items in list - O(M)
+		/// Total complexity - O(M)
+		/// # <weight>
+		#[weight = 1]
+		fn remove_signatory(origin, signatory: T::AccountId) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			<Accounts<T>>::mutate(&who, |opt| {
+				let multisig = opt.as_mut().ok_or(Error::<T>::UnknownMultisigAccount)?;
+				let pos = multisig.signatories.binary_search(&signatory)
+				    .map_err(|_| Error::<T>::NotInSignatories)?;
+				multisig.signatories.remove(pos);
+				Ok(())
+			})
+		}
+
+		/// Add a new signatory to the multisig account.
+		/// Can only be called by a multisig account.
+		///
+		/// TODO: update weights for `add_signatory`
+		/// # <weight>
+		/// Key: length of members in multisigConfig: M
+		/// - One storage read - O(1)
+		/// - search in members - O(M)
+		/// - Storage write - O(M)
+		/// Total complexity - O(M)
+		/// # <weight>
+		#[weight = 1]
+		fn add_signatory(origin, new_member: T::AccountId) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+            <Accounts<T>>::mutate(&who, |opt| {
```

Comment:
add tab here

Reply:
@humb1t yeah, I think it's because I used spaces instead of tabs